Ginger/GingerCore/ALM/Qtest/ImportFromQtest.cs

```diff
@@ -0,0 +1,1128 @@
+                    busFlow.Activities.Remove(activityToRemove);
+                }
+                var activityGroupsToRemove = busFlow.ActivitiesGroups.Where(x => x.ExternalID2 == tc.TestID).ToList();
+                foreach (ActivitiesGroup activityGroupToRemove in activityGroupsToRemove)
+                {
+                    activityGroupsToRemoveIndexes.Add(activityGroupToRemove.ExternalID2, busFlow.ActivitiesGroups.IndexOf(activityGroupToRemove));
+                }
+                foreach (ActivitiesGroup activityGroupToRemove in activityGroupsToRemove)
+                {
+                    busFlow.ActivitiesGroups.Remove(activityGroupToRemove);
+                }
+            }
+
+            int activityGroupToRemoveIndex;
+            foreach (QtestTest tc in tcsList)
+            {
+                activityGroupsToRemoveIndexes.TryGetValue(tc.TestID, out activityGroupToRemoveIndex);
+
+                //check if the TC is already exist in repository
+                ActivitiesGroup tcActivsGroup = new ActivitiesGroup();
+
+                tcActivsGroup = new ActivitiesGroup();
+                tcActivsGroup.Name = tc.TestName;
+                if (tc.LinkedTestID == null || tc.LinkedTestID == string.Empty)
+                {
+                    tcActivsGroup.ExternalID = tc.TestID;
+                    tcActivsGroup.ExternalID2 = tc.TestID;
+                }
+                else
+                {
+                    tcActivsGroup.ExternalID = tc.LinkedTestID;
+                    tcActivsGroup.ExternalID2 = tc.TestID; //original TC ID will be used for uploading the execution details back to QC
+                    tcActivsGroup.Description = tc.Description;
+                }
+                busFlow.AddActivitiesGroup(tcActivsGroup, activityGroupToRemoveIndex);
+
+                //Add the TC steps as Activities if not already on the Activities group
+                foreach (QtestTestStep step in tc.Steps)
+                {
+                    Activity stepActivity;
+                    bool toAddStepActivity = false;
+
+                    //check if mapped activity exist in repository
+                    Activity repoStepActivity = (Activity)GingerActivitiesRepo.Where(x => x.ExternalID == step.StepID).FirstOrDefault();
+                    if (repoStepActivity != null)
+                    {
+                        //check if it is part of the Activities Group
+                        ActivityIdentifiers groupStepActivityIdent = (ActivityIdentifiers)tcActivsGroup.ActivitiesIdentifiers.Where(x => x.ActivityExternalID == step.StepID).FirstOrDefault();
+                        if (groupStepActivityIdent != null)
+                        {
+                            //already in Activities Group so get link to it
+                            stepActivity = (Activity)busFlow.Activities.Where(x => x.Guid == groupStepActivityIdent.ActivityGuid).FirstOrDefault();
+                        }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)